src/FakeItEasy/Core/OrderedFakeAsserter.cs

```diff
@@ -2,15 +2,13 @@ namespace FakeItEasy.Core
 {
     using System;
     using System.Collections.Generic;
-    using System.Globalization;
-    using System.IO;
 
     internal class OrderedFakeAsserter : IFakeAsserter
     {
         private readonly IEnumerable<IFakeObjectCall> originalCallList;
         private readonly CallWriter callWriter;
         private readonly Queue<IFakeObjectCall> calls;
-        private readonly List<AssertedCall> assertedCalls = new List<AssertedCall>();
+        private readonly SynchronizedCollection<AssertedCall> assertedCalls = new SynchronizedCollection<AssertedCall>();

```

Comment:
I have reason to believe that #600 goes beyond just _literally that one exception that I reported_ and I'm not looking forward to run into another problem that's hard to troubleshoot. We obviously disagree on the need for _proactively_ solving these things, which is okay, but is a pretty fundamental difference of opinion given the changes _I, as the PR submitter and issue filer_, am interested in. I recognize not every change is interesting to the _project_ and while that's disappointing, I'll defer to your judgment as a project member.